src/middleware.py

```diff
@@ -21,7 +21,7 @@ def process_request(self, request):
             user, created = FacilityUser.objects.get_or_create(
                 username=self.username, facility=facility
             )
-            DevicePermissions.objects.create(
+            DevicePermissions.objects.get_or_create(
                 user=user, is_superuser=False, can_manage_content=True
```

Comment:
DevicePermissions is keyed by `user`, where the other fields are mutable. So, I think this will be more robust if we use `update_or_create` with those other fields in `defaults=` like this:

```suggestion
            DevicePermissions.objects.update_or_create(
                user=user, defaults={"is_superuser": False, "can_manage_content": True}
```

Alternatively, if we don't want to replace the existing DevicePermissions for a user, we can keep `get_or_create` but use `defaults=` like in the suggestion. That will avoid a potential mishap where a `DevicePermissions` entry already exists for the user but with different permissions.